feedthefox/users/views.py

```diff
@@ -1,17 +1,43 @@
 from django.contrib.auth.decorators import login_required
-from django.shortcuts import redirect, render
+from django.http import Http404
+from django.shortcuts import get_object_or_404, redirect, render
 
+from feedthefox.devices.models import DeviceInfo
 from feedthefox.users import forms
 
 
 @login_required
 def view_profile(request):
     newsletter_form = forms.ReceiveNewsLetterForm(request.POST or None,
-                                                  instance=request.user)
+                                                  instance=request.user,
+                                                  prefix='newsletter')
+    device_info_form = forms.UserDeviceInfoForm(request.POST or None,
+                                                prefix='device_info')
```

Comment:
Why don't you just check the POST request if `device` is present. If not then we have the newsletter form.